pkg/util/gendoc/gendoc.go

```diff
@@ -0,0 +1,143 @@
+package gendoc
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/cobra/doc"
+	"github.com/spf13/pflag"
+)
+
+const (
+	gendocTypeFlag = "type"
+
+	gendocMarkdown = "md"
+	gendocMan      = "man"
+
+	depthFlag     = "depth"
+	extensionFlag = "extension"
+)
+
+// Command returns command which generates user documentation for the argument.
+func Command(rootCmd *cobra.Command) *cobra.Command {
+	gendocCmd := &cobra.Command{
+		Use:   "gendoc <dir>",
+		Short: "Generate documentation for this command.",
+		Long: `Generate documentation for this command. If the template is not provided,
+builtin cobra generator is used and each subcommand is placed in
+a separate file in the same directory.
+
+The last optional argument specifies the template to use with text/template.
+In this case there is a number of helper functions which can be used:
+  replace STR FROM TO -- same as strings.ReplaceAll
+  join ARRAY SEPARATOR -- same as strings.Join
+  split STR SEPARATOR -- same as strings.Split
+  fullUse CMD -- slice of all command names starting from the parent
+  listFlags CMD -- list of command flags
+`,
+		RunE: func(cmd *cobra.Command, args []string) error {
```

Comment:
usually we tried not to use `RunE` in CLIs: https://github.com/nspcc-dev/neofs-node/issues/623

Reply:
created [issue](https://github.com/nspcc-dev/neofs-node/issues/1409)